bessctl/conf/testing/run_module_tests.bess

```diff
@@ -0,0 +1,166 @@
+    for i in range(num_output_ports):
+        sink = Sink()
+        module:i -> sink
+
+    time.sleep(5)
+
+
+## RUN TEST LOOP ##
+
+path = os.path.dirname(os.path.realpath(sys.argv[0]))
+for file in glob.glob(path + "/conf/testing/module_tests/*.py"):
+    bess.reset_all()
+    CRASH_TEST_INPUTS = []
+    OUTPUT_TEST_INPUTS = []
+    CUSTOM_TEST_FUNCTIONS = []
+
+    print("-- Running Module Tests for " + file + " --")
+    while True:  # wrap in this scoping to make sure variables don't leak into this script
+        exec(sugar.xform_file(file))
+        break
+
+    crash_testid = 1
+    for test in CRASH_TEST_INPUTS:
+        crash_test(test[0], test[1], test[2])
+        print("   " +
+              str(test[0]) +
+              " crash test " +
+              str(crash_testid) +
+              ": PASS")
+        crash_testid += 1
+
+    output_testid = 1
+    for test in OUTPUT_TEST_INPUTS:
+        input_ports = []
+        output_ports = []
+        sockets = []
+
+        for port_num in range(max(test[1], test[2])):
+            sockname = str(output_testid) + "moduletesting-" + \
+                SCRIPT_STARTTIME + str(port_num)
+            socket_port, mysocket = gen_socket_and_port(sockname)
+            input_ports.append(PortInc(port=sockname))
+            output_ports.append(PortOut(port=sockname))
+            sockets.append(mysocket)
+            if (port_num < test[1]):
+                input_ports[port_num] -> port_num:test[0]
+            if (port_num < test[2]):
+                test[0]:port_num -> output_ports[port_num]
+
+        bess.resume_all()
+
+        for test_case in test[3]:
+            input_port = test_case["input_port"]
+            output_port = test_case["output_port"]
+            input_pkt = test_case["input_packet"]
+            output_pkt = test_case["output_packet"]
+            if(input_pkt is not None):
+                sockets[input_port].send(str(input_pkt))
+            try:
+                return_pkt = scapy.Ether(sockets[output_port].recv(2048))
+            except socket.timeout:
+                return_pkt = None
+            if(not ((return_pkt is None and output_pkt is None) or
+                    (return_pkt == scapy.Ether(output_pkt)))):
+                print("Test failed!\n")
+                print("Input packet: " + pkt_str(input_pkt) + "\n")
+                print("Expected: " + pkt_str(output_pkt) + "\n")
+                print("Received: " + pkt_str(return_pkt) + "\n")
+                for sock in sockets:
+                    sock.close()
+                sys.exit(2)
+            print("   " +
+                  str(test[0]) +
+                  " output test " +
+                  str(output_testid) +
+                  ": PASS")
+            output_testid += 1
+        # tests passed!
+        for sock in sockets:
+            sock.close()
+        bess.pause_all()
+
+    for test in CUSTOM_TEST_FUNCTIONS:
+        result = test()
+        if(result != 0):
```

Comment:
How about capturing an exception rather than checking the return value of test functions, since:
* It's more pythonic.
* Usually unit test functions are expected to be void.
* The test function may fail to return a value due to unhandled exceptions. So you will need an exception handler here anyway.